### PR TITLE
Fixes #32933 - Upgrade to Pulpcore 3.14 in master

### PR DESCRIPTION
**Pull request by ianballou**

### CODE REVIEW

File: app/services/katello/pulp3/api/docker.rb [hunk 1]
```diff
@@ -37,7 +37,9 @@ def self.tag_image_class
         end
 
         def api_client
-          api_client_class(PulpContainerClient::ApiClient.new(smart_proxy.pulp3_configuration(PulpContainerClient::Configuration)))
+          config = smart_proxy.pulp3_configuration(PulpContainerClient::Configuration)
+          config.params_encoder = Faraday::FlatParamsEncoder
```
Comment (jlsherrill) [inline app/services/katello/pulp3/api/docker.rb:41]: mind adding a comment about this linking to https://pulp.plan.io/issues/8941  ?

File: app/services/katello/pulp3/api/docker.rb [hunk 2]
```diff
@@ -37,7 +37,9 @@ def self.tag_image_class
         end
 
         def api_client
-          api_client_class(PulpContainerClient::ApiClient.new(smart_proxy.pulp3_configuration(PulpContainerClient::Configuration)))
+          config = smart_proxy.pulp3_configuration(PulpContainerClient::Configuration)
```
Comment (ianballou) [inline app/services/katello/pulp3/api/docker.rb:40]: To fix this, Pulp will need to release all-new z-streams. I think we can live with this workaround for now. I'll make an issue when this PR merges.

**Review by jlsherrill — APPROVED** [review]
One small comment, but otherwise ack

**Comment by jlsherrill** [comment]
Going ahead and merging to push this train forward, but @ianballou continue testing with pulp-rpm 3.18.3 now that its releaesd

**Pull request merged**